server/graphql/v2/mutation/MemberMutations.js

```diff
@@ -0,0 +1,57 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { NotFound, Unauthorized } from '../../errors';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { Member } from '../object/Member';
+
+const memberMutations = {
+  editPublicMessage: {
+    type: new GraphQLNonNull(Member),
+    description: 'Edit the public message for the given Member of a Collective',
+    args: {
+      FromAccount: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the donating Collective',
+      },
+      ToAccount: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the receiving Collective',
+      },
+      message: {
+        type: GraphQLString,
+        description: 'New public message',
+      },
+    },
+    async resolve(_, args, req) {
+      const { FromAccount, ToAccount, message } = args;
+
+      const decodedToAccount = await fetchAccountWithReference(ToAccount);
+
+      if (!req.remoteUser.isAdminOfCollective(decodedToAccount)) {
+        throw new Unauthorized("You don't have the permission to edit this member's public message");
+      }
+
+      if (!decodedToAccount) {
+        throw new NotFound('Account not found');
+      }
+
+      const member = await models.Member.findOne({
+        where: {
+          MemberCollectiveId: FromAccount.legacyId,
```

Comment:
What if someone uses this mutation by passing an `id` rather than a `legacyId` in `FromAccount`?

We need to do the same `fetchAccountWithReference` for `fromAccount`.